frontend/src/metabase/nav/components/search/RecentsList/RecentsList.tsx

```diff
@@ -0,0 +1,164 @@
+import * as Urls from "metabase/lib/urls";
+import { SearchLoadingSpinner } from "metabase/nav/components/search/SearchResults/SearchResults";
+import { SearchResultLink } from "metabase/search/components/SearchResultLink";
+import type { RecentItem, UnrestrictedLinkEntity } from "metabase-types/api";
+import { useRecentItemListQuery } from "metabase/common/hooks";
+import RecentItems from "metabase/entities/recent-items";
+import { useDispatch } from "metabase/lib/redux";
+import { isSyncCompleted } from "metabase/lib/syncing";
+import { ItemIcon } from "metabase/search/components/SearchResult";
+import {
+  LoadingSection,
+  ModerationIcon,
+  ResultNameSection,
+  ResultTitle,
+  SearchResultContainer,
+} from "metabase/search/components/SearchResult/SearchResult.styled";
+import { Group, Loader, Stack, Title, Paper } from "metabase/ui";
+
+type RecentsListProps = {
+  onClick?: (elem: UnrestrictedLinkEntity) => void;
+  className?: string;
+};
+
+interface WrappedRecentItem extends RecentItem {
+  getUrl: () => string;
+  getIcon: () => {
+    name: IconName;
+    size?: number;
+    width?: number;
+    height?: number;
+  };
+}
+
+export const RecentsList = ({ onClick, className }: RecentsListProps) => {
+  const { data = [], isLoading: isRecentsListLoading } = useRecentItemListQuery(
+    {
+      reload: true,
+    },
+  );
+
+  const wrappedResults: WrappedRecentItem[] = useMemo(
+    () => data.map(item => RecentItems.wrapEntity(item)),
+    [data],
+  );
+
+  const dispatch = useDispatch();
+
+  const { getRef, cursorIndex } = useListKeyboardNavigation<
+    RecentItem,
+    HTMLButtonElement
+  >({
+    list: wrappedResults,
+    onEnter: (item: RecentItem) => onChangeLocation(item),
+  });
+
+  const onChangeLocation = (item: RecentItem) => {
+    const url = getItemUrl(item);
+    if (url) {
+      dispatch(push(url));
+    }
+  };
+
+  const onContainerClick = (item: RecentItem) => {
+    if (onClick) {
+      onClick({
+        ...item.model_object,
+        model: item.model,
+        name: item.model_object.display_name ?? item.model_object.name,
+        id: item.model_id,
+      });
+    } else {
+      onChangeLocation(item);
+    }
+  };
+
+  return (
+    <Paper withBorder className={className}>
+      {isRecentsListLoading ? (
+        <SearchLoadingSpinner />
+      ) : (
+        <Stack spacing="md" px="sm" py="md">
+          <Title order={4} px="sm">{t`Recently viewed`}</Title>
+          <Stack spacing={0}>
+            {wrappedResults.map((item, index) => {
+              const isActive = isItemActive(item);
+              const model = item.model;
+              const name = getItemName(item);
+              const moderated_status = getModeratedStatus(item);
+              const result = item;
+              const isSelected = cursorIndex === index;
+              const ref: Ref<HTMLButtonElement> = getRef(item) ?? null;
+
+              return (
```

Comment:
Could we reuse `SearchResult` here? It looks almost the same.
Or extract dumb UI component from it and reuse it here and there.

Reply:
Same with the weird pattern on the intermediate component - I just copied it over and there's just enough weirdness with the typing that would make this PR bigger than it already is

Reply:
I can do that in another PR, I have it somewhere